src/ATokenFlexVoting.sol

```diff
@@ -187,94 +195,102 @@ contract ATokenNaive is AToken {
-  /// @return `true` if the the previous balance of the user was 0
-  function _mintScaledWithCheckpoint(
-    address caller,
-    address onBehalfOf,
-    uint256 amount,
-    uint256 index
-  ) internal returns (bool) {
-    // We increment by `amount` instead of any computed/rebased amounts because
-    // `amount` is what actually gets transferred of the underlying asset. We
-    // need our checkpoints to still match up with underlying asset transactions.
-    Checkpoints.History storage _depositHistory = depositCheckpoints[onBehalfOf];
-    _depositHistory.push(_depositHistory.latest() + amount);
-    totalDepositCheckpoints.push(totalDepositCheckpoints.latest() + amount);
+  /// @notice Returns the _user's current balance in storage.
+  function _rawBalanceOf(address _user) internal view returns (uint256) {
+    return _userState[_user].balance;
+  }
 
-    return _mintScaled(caller, onBehalfOf, amount, index);
+  /// @notice Checkpoints the _user's current raw balance.
+  function _checkpointRawBalanceOf(address _user) internal {
+    balanceCheckpoints[_user].push(_rawBalanceOf(_user));
   }
 
-  function getPastDeposits(address _voter, uint256 _blockNumber) public returns (uint256) {
-    return depositCheckpoints[_voter].getAtBlock(_blockNumber);
+  /// @notice Returns the _user's balance in storage at the _blockNumber.
+  /// @param _user The account that's historical balance will be looked up.
+  /// @param _blockNumber The block at which to lookup the _user's balance.
+  function getPastStoredBalance(address _user, uint256 _blockNumber) public view returns (uint256) {
+    return balanceCheckpoints[_user].getAtProbablyRecentBlock(_blockNumber);
   }
 
-  function getPastTotalDeposits(uint256 _blockNumber) public returns (uint256) {
-    return totalDepositCheckpoints.getAtBlock(_blockNumber);
+  /// @notice Returns the total stored balance of all users at _blockNumber.
+  /// @param _blockNumber The block at which to lookup the total stored balance.
+  function getPastTotalBalances(uint256 _blockNumber) public view returns (uint256) {
+    return totalDepositCheckpoints.getAtProbablyRecentBlock(_blockNumber);
   }
 
   // forgefmt: disable-start
   //===========================================================================
   // BEGIN: Aave overrides
   //===========================================================================
-  /// Note: this has been modified from Aave v3's AToken to call our custom
-  /// mintScaledWithCheckpoint function.
+  /// Note: this has been modified from Aave v3's AToken to delegate voting
+  /// power to itself during initialization.
   ///
-  /// @inheritdoc IAToken
-  function mint(
-    address caller,
-    address onBehalfOf,
-    uint256 amount,
-    uint256 index
-  ) external virtual override onlyPool returns (bool) {
-    return _mintScaledWithCheckpoint(caller, onBehalfOf, amount, index);
+  /// @inheritdoc AToken
+  function initialize(
```

Comment:
cc @mds1 @apbendi

Reply:
My gut reaction is to agree with @wildmolasses but I don't feel strongly and I don't think it's worth bikeshedding too much about. Unless @wildmolasses (which I'm guessing he doesn't since he used `nit`) I'd say just go with whatever you'd prefer @davidlaprade.

Reply:
Thanks. Going to punt for now then, ticket tracking this is here: https://github.com/ScopeLift/flexible-voting/issues/26